massa-network/src/peer_info_database.rs

```diff
@@ -321,169 +407,40 @@ impl PeerInfoDatabase {
-            res_ips.extend(
-                sorted_peers
-                    .into_iter()
-                    .take(available_slots_bootstrap)
-                    .map(|p| p.ip),
-            )
-        }
-
-        if available_slots_nonbootstrap > 0 {
-            let now = MassaTime::compensated_now(self.clock_compensation)?;
-            let mut sorted_peers: Vec<PeerInfo> = self
-                .peers
-                .values()
-                .filter(|&p| {
-                    if !(!p.bootstrap && p.advertised && !p.banned && !p.is_active()) {
-                        return false;
-                    }
-                    if let Some(last_failure) = p.last_failure {
-                        if let Some(last_alive) = p.last_alive {
-                            if last_alive > last_failure {
-                                return true;
-                            }
-                        }
-                        return now
-                            .saturating_sub(last_failure)
-                            .saturating_sub(self.wakeup_interval)
-                            > MassaTime::from(0u64);
-                    }
-                    true
-                })
-                .copied()
-                .collect();
-            sorted_peers
-                .sort_unstable_by_key(|&p| (p.last_failure, std::cmp::Reverse(p.last_alive)));
-            res_ips.extend(
-                sorted_peers
-                    .into_iter()
-                    .take(available_slots_nonbootstrap)
-                    .map(|p| p.ip),
-            )
-        }
-
-        Ok(res_ips)
+        self.request_dump()
     }
 
-    /// returns Hashmap of ipAddrs -> Peerinfo
-    pub fn get_peers(&self) -> &HashMap<IpAddr, PeerInfo> {
-        &self.peers
-    }
+    ////////////////////////////////
+    // high level peer management //
+    ////////////////////////////////
 
-    /// Returns a vec of advertisable IpAddrs sorted by ( last_failure, rev(last_success) )
-    pub fn get_advertisable_peer_ips(&self) -> Vec<IpAddr> {
-        let mut sorted_peers: Vec<PeerInfo> = self
-            .peers
-            .values()
-            .filter(|&p| (p.advertised && !p.banned))
-            .copied()
-            .collect();
-        sorted_peers.sort_unstable_by_key(|&p| (std::cmp::Reverse(p.last_alive), p.last_failure));
-        let mut sorted_ips: Vec<IpAddr> = sorted_peers
-            .into_iter()
-            .take(MAX_ADVERTISE_LENGTH as usize)
-            .map(|p| p.ip)
-            .collect();
-        if let Some(our_ip) = self.network_settings.routable_ip {
-            sorted_ips.insert(0, our_ip);
-            sorted_ips.truncate(MAX_ADVERTISE_LENGTH as usize);
+    /// Unbans a list of ip
+    pub async fn unban(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
+        for ip in ips.into_iter() {
+            if let Some(peer) = self.peers.get_mut(&ip) {
+                peer.peer_type = Default::default();
```

Comment:
why is unbanning changing the peer type ?

Reply:
It's an artifact. I tried having Banned as a peerType variant, it was much more painful than I expected.